InEenNotendop.Business/MidiProcessor.cs

```diff
@@ -103,29 +104,44 @@ private void InitializeMidi(string desiredOutDevice)
 
             var numDevices = MidiIn.NumberOfDevices;
             Debug.WriteLine($"numDevices: {numDevices}");
-            var desiredDeviceIndex = 0; // DEZE KAN VERANDEREN SOMS SPONTAAN
-            if(numDevices < 0)
-            {
-                numDevices = 0;
-            }
-            if (desiredDeviceIndex < numDevices)
+            var deviceIndex = MidiUtilities.FindMidiDevice("Impact GX49", InOrOut.IN);
+
+            if (deviceIndex != -1)
             {
                 try
                 {
-                    _midiIn?.Dispose();
-                    _midiIn = new MidiIn(desiredDeviceIndex);
+                    _midiIn = new MidiIn(deviceIndex);
                     _midiIn.MessageReceived += MidiInMessageReceived;
                     _midiIn.Start();
-                } catch (NAudio.MmException e)
+                }
+                catch (Exception ex)
                 {
-                    Debug.WriteLine($"Error: {e.Message}");
+                    Debug.WriteLine($"Error: {ex.Message}");
                 }
 
             }
             else
             {
-                MidiDeviceNotFound?.Invoke(this);
+                throw new Exception($"Device with desired device index {deviceIndex} not found");
             }
+            //if (desiredDeviceIndex < numDevices)
```

Comment:
WAAROM STAAT HIER WEG GESLASHED CODE REEEEE

Reply:
is grappig

Reply:
vorige keer ging docent zeiken over oude kunnen niet kunnen zien, nu niets meer verwijderen maar gwn wegcommenten :)

Reply:
ze kunnen hooguit een paar punten weg nemen voor weg geslashed comment, dus beter teveel // dan dubbele code

Reply:
uitroepteken

Reply:
hoe gaat het?

Reply:
waarom reageer je niet

Reply:
:(

Reply:
krijg je hier ook mailjtes van?

Reply:
email

Reply:
![image](https://github.com/danieltoof/Piano/assets/105288240/e23f7f1a-b45c-432d-a686-7bf651b87a60)